logviewer/src/main/java/org/togetherjava/tjbot/logwatcher/Application.java

```diff
@@ -34,9 +35,12 @@
 @Push
 public class Application extends SpringBootServletInitializer implements AppShellConfigurator {
 
+    @SuppressWarnings({"java:S2387"})
```

Comment:
What is this suppress doing here? I dont like it and would remove it. But if you want to keep it, put a comment explaining what it is about.

Reply:
The `SpringBootServletInitializer` class has a protected field called `logger` and sonar lint is unhappy that the field names are same.

Reply:
then dont ignore the warning but rename the field lol.

Reply:
or, if springs logger is basically the logger already, **reuse it** instead of creating a new logger.

Reply:
The springs logger is from the apache logging framework, also renaming this field would lose consistency which is against the whole idea of this PR.

Reply:
> or, if springs logger is basically the logger already, **reuse it** instead of creating a new logger.

It's our code, not spring's code. So I 100% disagree

If we have an error, and the error is in our code, it should be logged in OUR code, not Spring's

Reply:
> then dont ignore the warning but rename the field lol.

Inconsistency, no

Reply:
> It's our code, not spring's code. So I 100% disagree
It kinda depends on the purpose of this field. If this is a protected field meant to be used by implementing classes and if it follows our existing logging configuration already, then it might make sense.

Other than that I agree with you and we should rename our field to get rid of the name clash and confusion.

Reply:
> > then dont ignore the warning but rename the field lol.
> 
> Inconsistency, no

Consistency is not an excuse to write bad code. Hiding **visible** fields is a bad idea, causes bugs and confusion.

Reply:
> Technically we could reuse that logger, but it is from the apache loggings framework, I hope that is ok.

What does its javadoc say? Its protected/public on purpose. So I would assume that they explain when and how to use it in their Javadoc.

Reply:
> What does its javadoc say?

There is no javadoc

```java
public abstract class SpringBootServletInitializer implements WebApplicationInitializer {

	protected Log logger; // Don't initialize early

	private boolean registerErrorPageFilter = true;
	...
}
```

Reply:
Wait, wtf have we been thinking 🤦 It is a non-static field, we cannot use it in the `main` method.

Reply:
> Wait, wtf have we been thinking 🤦 It is a non-static field, we cannot use it in the main method.

We are still hiding it, which we should not. We are extending this class after all. Imagine you would have a super class
```java
class Animal {
  protected String name;
  ...
}
```
then you also wouldnt start doing
```java
class Dog extends Animal {
  private static final String name = "Foo";
  ...
}
```
but call it differently to make clear what differentiates it from the `name` that you have inside the application already.

Otherwise people are deeply confused when for example using **inside a non-static Dog method** `name`, the call is ambigious then to a reader.

Reply:
We never call `logger` in a non-static context though. I think it should be fine.

Reply:
@java-coding-prodigy Just call it `applicationLogger` please - gets rid of the name hiding, makes its purpose clear (or rather who owns it) and everyone is happy.

Reply:
@java-coding-prodigy could u finish this quickly when u have time? :) or should someone take over?

Reply:
Really sorry, I had quite a lot of work to do(I still am busy, but can do this). Working on it rn.

Reply:
No worries at all. Take your time.

Reply:
Done 👍